Migrate log4cats artifacts

fthomas opened the pull request:
I was just made aware that log4cats is now published under `org.typelevel`.  This will rewrite the organization of log4cats-core, log4cats-noop, log4cats-slf4j, and log4cats-testing. The package names for these were also changed from `io.chrisdavenport.log4cats` to `org.typelevel.log4cats`, so if we merge this all update PRs will fail because of that source incompatibility. In theory this could be salvaged with a Scalafix migration that rewrites the imports and is used at the same time, but I'm not sure if it is worth the hassle.

/cc @ChristopherDavenport, @rossabaker

rossabaker commented:
I solved it with a search and replace when upgrading http4s.  I think it's 100% source compatible but for the package name.

They're less prominent, but unique, vault and keypool all made the same organization and package move.

fthomas commented:
I'm now trying to write a Scalafix. Maybe it can be done in a few lines of code. Let's see.

fthomas commented:
I added a Scalafix and it worked: https://github.com/scala-steward-org/scala-steward/pull/1969 :tada:

The PR changed the organization in the build and updated all imports.
